### PR TITLE
Fix #5473 Trigger push webhook on new branch

### LINKED ISSUE #5473: Webhook not triggering when new branch pushed to mirrored repository
- Gogs version (or commit ref): 0.11.66
- Git version: 
- Operating system: 
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gogs.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist (usually found in `log/gogs.log`):

## Description

Webhook is not triggering after sync when new branch is pushed to mirrored repository.
I believe that this piece of code is responsible: https://github.com/gogs/gogs/blob/a1098384c09bc0d569ec82d19c26415e00cc364b/models/mirror.go#L406-L412

Expected behaviour would be to trigger webhooks on all changes. However, current implementation triggers webhook on new branches only on second push.

...

### PR DESCRIPTION
**Pull request by danilodemorais**
This commit fixes issue #5473 and makes a new branch behave like a
      push event and trigger the appropriate webhook.

### CODE REVIEW

File: models/mirror.go [hunk 1]
```diff
@@ -403,14 +403,6 @@ func SyncMirrors() {
 				continue
 			}
 
-			// Create reference
-			if result.oldCommitID == GIT_SHORT_EMPTY_SHA {
-				if err = MirrorSyncCreateAction(m.Repo, result.refName); err != nil {
-					log.Error(2, "MirrorSyncCreateAction [repo_id: %d]: %v", m.RepoID, err)
-				}
-				continue
```
Comment (Unknwon) [inline models/mirror.go]: I think all we need is just to remove this single line of "continue", everything else should works automatically.

File: models/mirror.go [hunk 2]
```diff
@@ -419,21 +411,51 @@ func SyncMirrors() {
...
+				if err != nil {
+					log.Error(2, "GetFullCommitID [%d]: %v", m.RepoID, err)
+					continue
+				}
+				if commit, err := gitRepo.GetCommit(refUniqueCommitID); err != nil {
+					log.Error(2, "GetCommit [repo_id: %d, commit_id: %s]: %v", m.RepoID, refUniqueCommitID, err)
+					continue
+				} else {
+					commits = list.New()
+					commits.PushFront(commit)
```
Comment (Unknwon) [inline models/mirror.go]: This only collects the latest commit, which is not always true, a branch can be created with m multiple commits. (see my another comment)

File: models/mirror.go (hunk 1 above)
Comment (danilodemorais) [reply models/mirror.go]: If the continue is removed then the code will fail later when trying to retrieve the full commitID for GIT_SHORT_EMPTY_SHA which does not exist.

File: models/mirror.go (hunk 2 above)
Comment (danilodemorais) [reply models/mirror.go]: I'm working on it

**Pull request closed without merging**